### ISSUE TITLE
Tracking the initial render of Index.tsx

### ISSUE THREAD

**Issue by altenstedt** [anastasiia-zolochevska/react-appinsights#43]
It seems like the documentation in the README file for tracking router changes does not track the request to the initial rendering of Index.tsx.  What is the best way to handle that?

**Comment by iulixxi** [comment]
I have the same issue... noticed that it does not register the entry point at all...
So basically if you use the browser URL bar and navigate to a different page it does not record that ... any ideas?

**Comment by pviotti** [comment]
Good catch. `react-appinsights` relies on `history` to track page changes. Unfortunately, by design, the `history.listen` callback is not fired on the first page load (see [this ](https://github.com/ReactTraining/history/issues/479#issuecomment-307544999) comment).  
So the workaround would be recording the page view manually, by using react-appinsights's internal instance of AppInsights. Here's an example excerpt of the index.tsx (using the react-appinsights v3.x ):  

```javascript 

import React from 'react';
import ReactDOM from 'react-dom';
import App from './App';
import { Router } from 'react-router-dom';
import { ReactAI } from 'react-appinsights';
import { createBrowserHistory } from "history"

const history = createBrowserHistory()
ReactAI.initialize({ instrumentationKey: "InstrumentationKey", history: history, debug: true });

const pageViewTelemetry = { uri: history.location.pathname };
ReactAI.rootInstance.trackPageView(pageViewTelemetry);
console.log("Recording initial page view.", `uri: ${history.location.pathname}`);

ReactDOM.render((
    <Router history={history}>
        <App />
    </Router>),
    document.getElementById('root'));
```

(I'll keep this open to remember to write a test and document this behaviour)

**Comment by pviotti** [comment]
This should be solved in version 3.0.0-beta.5, just released. To install it, issue `npm install --save react-appinsights@beta` (docs and code are in the typescript-rewrite branch). As soon as AppInsights releases their new SDK, this will become the new stable version.

**Comment by altenstedt** [comment]
I can confirm that the latest beta does indeed track the initial request using the following initialization:

```
const history = createBrowserHistory();
ReactAI.initialize({ 
  instrumentationKey: appInsightsInstrumentationKey,
  history: history,
  initialContext: { ... } 
});
```
